c++/src/capnp/compat/json.c++

```diff
@@ -237,9 +238,20 @@ void JsonCodec::encode(DynamicValue::Reader input, Type type, JsonValue::Builder
     case schema::Type::UINT8:
     case schema::Type::UINT16:
     case schema::Type::UINT32:
+      output.setNumber(input.as<double>());
+      break;
     case schema::Type::FLOAT32:
     case schema::Type::FLOAT64:
-      output.setNumber(input.as<double>());
+      {
+        double value = input.as<double>();
+        if (std::isinf(value) || std::isnan(value)) {
```

Comment:
Such NIH, wow. I had to introduce `kj::isNaN()` to do so. I'll send another tiny PR after this is merged to dedup its uses elsewhere.